### PR TITLE
LPS-109049 portal-search-web: Add ADTs for sort widget

### PR DESCRIPTION
**Pull request by oliv-yu**
https://issues.liferay.com/browse/LPS-109049

Sort widget!

@jonmak08

### CODE REVIEW

**Comment by oliv-yu** [comment]
ci:test:sf:norebase

**Comment by oliv-yu** [comment]
ci:test:search:norebase

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/sort/display/context/SortDisplayBuilder.java [hunk 1]
```diff
@@ -70,6 +96,9 @@ protected SortTermDisplayContext buildTermDisplayContext(
 			new SortTermDisplayContext();
 
 		sortTermDisplayContext.setLabel(label);
+		sortTermDisplayContext.setLanguageLabel(
+			LanguageUtil.get(
+				PortalUtil.getHttpServletRequest(_renderRequest), label));
```
Comment (thektan) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/sort/display/context/SortDisplayBuilder.java]: We're trying to move away from these `*Util` classes. Can we try using the `Language` and `Portal` classes directly instead?

```java
language.get(portal.getHttpServletRequest(_renderRequest), label));
```

```java
@Reference
protected Language language;

@Reference
protected Portal portal;
```

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/sort/template/SortPortletDisplayTemplateHandler.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import java.util.Locale;
+import java.util.Map;
+import java.util.ResourceBundle;
+
+import org.osgi.service.component.annotations.Component;
+import org.osgi.service.component.annotations.ConfigurationPolicy;
+import org.osgi.service.component.annotations.Reference;
+
+/**
+ * @author Kevin Tan
```
Comment (thektan) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/sort/template/SortPortletDisplayTemplateHandler.java]:
```suggestion
 * @author Olivia Yu
```

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/sort/configuration/SortPortletInstanceConfiguration.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ */
+
+package com.liferay.portal.search.web.internal.sort.configuration;
+
+import aQute.bnd.annotation.metatype.Meta;
+
+import com.liferay.portal.configuration.metatype.annotations.ExtendedObjectClassDefinition;
+
+/**
+ * @author Kevin Tan
```
Comment (thektan) [inline modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/sort/configuration/SortPortletInstanceConfiguration.java]:
```suggestion
 * @author Olivia Yu
```

File: modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/sort/display/context/SortDisplayBuilder.java (hunk 1 above)
Comment (oliv-yu) [reply modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/sort/display/context/SortDisplayBuilder.java]: Doing this seems to cause a null pointer exception for me :\

Comment (thektan) [reply modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/sort/display/context/SortDisplayBuilder.java]: Ah, I think we'll have to pass these in then. In `SortPortlet.java` and `SortConfigurationAction.java` we can add:
```java
@Reference
protected Language language;

@Reference
protected Portal portal;
```

Then the `SortDisplayBuilder` will be:
```java
SortDisplayBuilder(language, portal, renderRequest, sortPortletPreferences)
```

Comment (oliv-yu) [reply modules/apps/portal-search/portal-search-web/src/main/java/com/liferay/portal/search/web/internal/sort/display/context/SortDisplayBuilder.java]: Ooo thank you! That worked

**Comment by thektan** [comment]
ci:test:sf

**Comment by thektan** [comment]
ci:test:search:norebase

**Comment by thektan** [comment]
Pull request submitted to https://github.com/brianchandotcom/liferay-portal/pull/85680. [See changes here.](https://github.com/thektan/liferay-portal/compare/c5f6d0c9443f~1...c5f6d0c9443f)

**Pull request closed without merging**